problems when compile the xcikit project

# Prepare build directory
mkdir build && cd build

# Install dependencies using Conan.
conan install .. --build missing

# Configure
cmake .. -G Ninja -DCMAKE_INSTALL_PREFIX=~/sdk/xcikit


all that is ok,but when i build use the cmd "cmake --build ." FAILED


D:\xcikit-master\build>cmake --build .
[1/168] Building CXX object src\xci\core\CMakeFiles\xci-core.dir\cmake_pch.cxx.obj
FAILED: src/xci/core/CMakeFiles/xci-core.dir/cmake_pch.cxx.obj
C:\PROGRA~2\MICROS~2\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP -DNOMINMAX -D_CRT_NONSTDC_NO_WARNINGS -D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES=1 -D_CRT_SECURE_NO_WARNINGS -D_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING -D_SILENCE_CXX17_STRSTREAM_DEPRECATION_WARNING -I..\src -Iinclude -I..\third_party\widechar_width\.. /DWIN32 /D_WINDOWS /EHsc /Zi /Ob0 /Od /RTC1 -MDd /W4 /wd4100 /wd4146 /wd4200 /wd4244 /wd4702 /wd5105 /permissive- /Zc:preprocessor /Zc:inline /Zc:__cplusplus /utf-8 /bigobj -std:c++latest /YcD:/xcikit-master/build/src/xci/core/CMakeFiles/xci-core.dir/cmake_pch.hxx /FpD:/xcikit-master/build/src/xci/core/CMakeFiles/xci-core.dir/./cmake_pch.cxx.pch /FID:/xcikit-master/build/src/xci/core/CMakeFiles/xci-core.dir/cmake_pch.hxx /showIncludes /Fosrc\xci\core\CMakeFiles\xci-core.dir\cmake_pch.cxx.obj /Fdsrc\xci\core\xci-core.dir\xci-core.pdb /FS -c src\xci\core\CMakeFiles\xci-core.dir\cmake_pch.cxx
D:/xcikit-master/build/src/xci/core/CMakeFiles/xci-core.dir/cmake_pch.hxx(5): fatal error C1083: 无法打开包括文件: “fmt/format.h”: No such file or directory
[2/168] Building CXX object src\xci\data\CMakeFiles\xci-data.dir\Crc32.cpp.obj
FAILED: src/xci/data/CMakeFiles/xci-data.dir/Crc32.cpp.obj
C:\PROGRA~2\MICROS~2\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP -DNOMINMAX -D_CRT_NONSTDC_NO_WARNINGS -D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES=1 -D_CRT_SECURE_NO_WARNINGS -D_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING -D_SILENCE_CXX17_STRSTREAM_DEPRECATION_WARNING -I..\src -Iinclude -ID:\boost_1_81_0 /DWIN32 /D_WINDOWS /EHsc /Zi /Ob0 /Od /RTC1 -MDd /W4 /wd4100 /wd4146 /wd4200 /wd4244 /wd4702 /wd5105 /permissive- /Zc:preprocessor /Zc:inline /Zc:__cplusplus /utf-8 /bigobj -std:c++latest /showIncludes /Fosrc\xci\data\CMakeFiles\xci-data.dir\Crc32.cpp.obj /Fdsrc\xci\data\CMakeFiles\xci-data.dir\xci-data.pdb /FS -c ..\src\xci\data\Crc32.cpp
..\src\xci\data\Crc32.cpp(8): fatal error C1083: 无法打开包括文件: “zlib.h”: No such file or directory
[3/168] Building CXX object examples\core\CMakeFiles\demo_termctl.dir\demo_termctl.cpp.obj
FAILED: examples/core/CMakeFiles/demo_termctl.dir/demo_termctl.cpp.obj
C:\PROGRA~2\MICROS~2\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP -DNOMINMAX -D_CRT_NONSTDC_NO_WARNINGS -D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES=1 -D_CRT_SECURE_NO_WARNINGS -D_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING -D_SILENCE_CXX17_STRSTREAM_DEPRECATION_WARNING -I..\examples -I..\src -Iinclude /DWIN32 /D_WINDOWS /EHsc /Zi /Ob0 /Od /RTC1 -MDd /W4 /wd4100 /wd4146 /wd4200 /wd4244 /wd4702 /wd5105 /permissive- /Zc:preprocessor /Zc:inline /Zc:__cplusplus /utf-8 /bigobj -std:c++latest /showIncludes /Foexamples\core\CMakeFiles\demo_termctl.dir\demo_termctl.cpp.obj /Fdexamples\core\CMakeFiles\demo_termctl.dir\ /FS -c ..\examples\core\demo_termctl.cpp
D:\xcikit-master\src\xci/core/TermCtl.h(10): fatal error C1083: 无法打开包括文件: “fmt/format.h”: No such file or directory
[4/168] Building CXX object src\xci\data\CMakeFiles\xci-data.dir\Schema.cpp.obj
FAILED: src/xci/data/CMakeFiles/xci-data.dir/Schema.cpp.obj
C:\PROGRA~2\MICROS~2\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP -DNOMINMAX -D_CRT_NONSTDC_NO_WARNINGS -D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES=1 -D_CRT_SECURE_NO_WARNINGS -D_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING -D_SILENCE_CXX17_STRSTREAM_DEPRECATION_WARNING -I..\src -Iinclude -ID:\boost_1_81_0 /DWIN32 /D_WINDOWS /EHsc /Zi /Ob0 /Od /RTC1 -MDd /W4 /wd4100 /wd4146 /wd4200 /wd4244 /wd4702 /wd5105 /permissive- /Zc:preprocessor /Zc:inline /Zc:__cplusplus /utf-8 /bigobj -std:c++latest /showIncludes /Fosrc\xci\data\CMakeFiles\xci-data.dir\Schema.cpp.obj /Fdsrc\xci\data\CMakeFiles\xci-data.dir\xci-data.pdb /FS -c ..\src\xci\data\Schema.cpp
D:\xcikit-master\src\xci\data\ArchiveBase.h(13): fatal error C1083: 无法打开包括文件: “fmt/core.h”: No such file or directory
[5/168] Building CXX object src\xci\data\CMakeFiles\xci-data.dir\BinaryReader.cpp.obj
FAILED: src/xci/data/CMakeFiles/xci-data.dir/BinaryReader.cpp.obj
C:\PROGRA~2\MICROS~2\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP -DNOMINMAX -D_CRT_NONSTDC_NO_WARNINGS -D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES=1 -D_CRT_SECURE_NO_WARNINGS -D_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING -D_SILENCE_CXX17_STRSTREAM_DEPRECATION_WARNING -I..\src -Iinclude -ID:\boost_1_81_0 /DWIN32 /D_WINDOWS /EHsc /Zi /Ob0 /Od /RTC1 -MDd /W4 /wd4100 /wd4146 /wd4200 /wd4244 /wd4702 /wd5105 /permissive- /Zc:preprocessor /Zc:inline /Zc:__cplusplus /utf-8 /bigobj -std:c++latest /showIncludes /Fosrc\xci\data\CMakeFiles\xci-data.dir\BinaryReader.cpp.obj /Fdsrc\xci\data\CMakeFiles\xci-data.dir\xci-data.pdb /FS -c ..\src\xci\data\BinaryReader.cpp
D:\xcikit-master\src\xci\data\ArchiveBase.h(13): fatal error C1083: 无法打开包括文件: “fmt/core.h”: No such file or directory
[6/168] Building CXX object src\xci\data\CMakeFiles\xci-data.dir\BinaryWriter.cpp.obj
FAILED: src/xci/data/CMakeFiles/xci-data.dir/BinaryWriter.cpp.obj
C:\PROGRA~2\MICROS~2\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP -DNOMINMAX -D_CRT_NONSTDC_NO_WARNINGS -D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES=1 -D_CRT_SECURE_NO_WARNINGS -D_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING -D_SILENCE_CXX17_STRSTREAM_DEPRECATION_WARNING -I..\src -Iinclude -ID:\boost_1_81_0 /DWIN32 /D_WINDOWS /EHsc /Zi /Ob0 /Od /RTC1 -MDd /W4 /wd4100 /wd4146 /wd4200 /wd4244 /wd4702 /wd5105 /permissive- /Zc:preprocessor /Zc:inline /Zc:__cplusplus /utf-8 /bigobj -std:c++latest /showIncludes /Fosrc\xci\data\CMakeFiles\xci-data.dir\BinaryWriter.cpp.obj /Fdsrc\xci\data\CMakeFiles\xci-data.dir\xci-data.pdb /FS -c ..\src\xci\data\BinaryWriter.cpp
D:\xcikit-master\src\xci\data\ArchiveBase.h(13): fatal error C1083: 无法打开包括文件: “fmt/core.h”: No such file or directory
[7/168] Building CXX object src\xci\data\CMakeFiles\xci-data.dir\Dumper.cpp.obj
FAILED: src/xci/data/CMakeFiles/xci-data.dir/Dumper.cpp.obj
C:\PROGRA~2\MICROS~2\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP -DNOMINMAX -D_CRT_NONSTDC_NO_WARNINGS -D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES=1 -D_CRT_SECURE_NO_WARNINGS -D_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING -D_SILENCE_CXX17_STRSTREAM_DEPRECATION_WARNING -I..\src -Iinclude -ID:\boost_1_81_0 /DWIN32 /D_WINDOWS /EHsc /Zi /Ob0 /Od /RTC1 -MDd /W4 /wd4100 /wd4146 /wd4200 /wd4244 /wd4702 /wd5105 /permissive- /Zc:preprocessor /Zc:inline /Zc:__cplusplus /utf-8 /bigobj -std:c++latest /showIncludes /Fosrc\xci\data\CMakeFiles\xci-data.dir\Dumper.cpp.obj /Fdsrc\xci\data\CMakeFiles\xci-data.dir\xci-data.pdb /FS -c ..\src\xci\data\Dumper.cpp
D:\xcikit-master\src\xci\data\ArchiveBase.h(13): fatal error C1083: 无法打开包括文件: “fmt/core.h”: No such file or directory
[8/168] Building CXX object examples\core\CMakeFiles\demo_chunked_stack.dir\demo_chunked_stack.cpp.obj
FAILED: examples/core/CMakeFiles/demo_chunked_stack.dir/demo_chunked_stack.cpp.obj
C:\PROGRA~2\MICROS~2\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP -DNOMINMAX -D_CRT_NONSTDC_NO_WARNINGS -D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES=1 -D_CRT_SECURE_NO_WARNINGS -D_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING -D_SILENCE_CXX17_STRSTREAM_DEPRECATION_WARNING -I..\examples -I..\src -Iinclude /DWIN32 /D_WINDOWS /EHsc /Zi /Ob0 /Od /RTC1 -MDd /W4 /wd4100 /wd4146 /wd4200 /wd4244 /wd4702 /wd5105 /permissive- /Zc:preprocessor /Zc:inline /Zc:__cplusplus /utf-8 /bigobj -std:c++latest /showIncludes /Foexamples\core\CMakeFiles\demo_chunked_stack.dir\demo_chunked_stack.cpp.obj /Fdexamples\core\CMakeFiles\demo_chunked_stack.dir\ /FS -c ..\examples\core\demo_chunked_stack.cpp
D:\xcikit-master\src\xci/core/log.h(12): fatal error C1083: 无法打开包括文件: “fmt/format.h”: No such file or directory
[9/168] Building CXX object examples\core\CMakeFiles\demo_event.dir\demo_event.cpp.obj
FAILED: examples/core/CMakeFiles/demo_event.dir/demo_event.cpp.obj
C:\PROGRA~2\MICROS~2\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP -DNOMINMAX -D_CRT_NONSTDC_NO_WARNINGS -D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES=1 -D_CRT_SECURE_NO_WARNINGS -D_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING -D_SILENCE_CXX17_STRSTREAM_DEPRECATION_WARNING -I..\examples -I..\src -Iinclude /DWIN32 /D_WINDOWS /EHsc /Zi /Ob0 /Od /RTC1 -MDd /W4 /wd4100 /wd4146 /wd4200 /wd4244 /wd4702 /wd5105 /permissive- /Zc:preprocessor /Zc:inline /Zc:__cplusplus /utf-8 /bigobj -std:c++latest /showIncludes /Foexamples\core\CMakeFiles\demo_event.dir\demo_event.cpp.obj /Fdexamples\core\CMakeFiles\demo_event.dir\ /FS -c ..\examples\core\demo_event.cpp
D:\xcikit-master\src\xci/core/log.h(12): fatal error C1083: 无法打开包括文件: “fmt/format.h”: No such file or directory

////////////////////////////////////////////////////////////////////////////////////
i don't know where the error occur, i use conan manage all packages, also fmt is installed in the directory of "C:\Users\rt\.conan\data\.."

sorry my poor english.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -115,3 +115,4 @@ workflows:
       - conan_package:
           name: Ubuntu 22.04 - Clang 14 - Conan package
           image: ubuntu
+          build-args: -o xcikit:tools=False -o xcikit:examples=False -o xcikit:tests=False -o xcikit:benchmarks=False

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,10 @@
 # 3.16.3 in Emscripten Docker image
 # 3.16 introduced CMAKE_MESSAGE_INDENT
 cmake_minimum_required(VERSION 3.16...3.25)
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+include(XciBuildType)
+
 file(STRINGS "VERSION" XCI_VERSION)
 project(xcikit
     LANGUAGES CXX
@@ -65,7 +69,7 @@ option(XCI_DEBUG_MARKUP_DUMP_TOKENS "Text markup parser debugging." OFF)
 set(XCI_CXX_STANDARD 20)
 
 set(XCIKIT_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
-list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
 if (CMAKE_CROSSCOMPILING)
     # Needed for Emscripten to find packages in build dir
     list(APPEND CMAKE_FIND_ROOT_PATH "/")

--- a/CONTRIBUTING.adoc
+++ b/CONTRIBUTING.adoc
@@ -1,0 +1,21 @@
+= Contributing to xcikit
+
+First, note that this is a personal hobby project. It's basically my playground for C++ and graphics stuff.
+Sometimes I rewrite things. Sometimes I drop things I no longer like. Some parts might be quite broken at times.
+
+== You've found a bug
+
+Please report an issue, or submit a fix as a pull request. Bug fixes are welcome!
+
+== You want to add a feature
+
+You're of course free to fork this project, and develop your feature.
+If you want it merged eventually, it might be better to discuss it first.
+
+== By submitting a pull request
+
+* You confirm that you are the author of the code and that you have the right to license the contribution to the xcikit project owner.
+
+* You confirm that your contribution is not bound by any software patents or other means that could conflict with licensing under current or any future xcikit license.
+
+* You agree that your contribution is licensed under the xcikit license, and you agree to any future changes to the licensing.

--- a/README.adoc
+++ b/README.adoc
@@ -410,6 +410,17 @@ Run the build (only 'core' and 'script' components work at this time):
 ./build.sh --debug --emscripten --profile emscripten core script
 ----
 
+[#package]
+=== Building Conan package
+
+The defaults in the Conan recipe (`conanfile.py`) are meant to make it easier for consumers (`conan install`) than packagers. That means, the tests and examples are built and packaged by default. Add the following options to build a package with only the development libs:
+
+[source,bash]
+----
+conan create . --build=missing -o xcikit:tools=False -o xcikit:examples=False -o xcikit:tests=False -o xcikit:benchmarks=False
+----
+
+
 == How to use in a client program
 
 === Linking with a client using only CMake

--- a/cmake/XciBuildOptions.cmake
+++ b/cmake/XciBuildOptions.cmake
@@ -15,6 +15,13 @@ option(FAIL_ON_WARNINGS "Treat warnings as errors. Applies also to clang-tidy." 
 option(ENABLE_TIDY "Run clang-tidy on each compiled file, when available." OFF)
 option(ENABLE_IWYU "Run iwyu (Include What You Use) on each compiled file, when available." OFF)
 
+# default build type + possible values for cmake gui
+if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build." FORCE)
+endif()
+set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+message(STATUS "Build type: ${CMAKE_BUILD_TYPE}")
+
 # optimizations
 if(CMAKE_BUILD_TYPE STREQUAL "MinSizeRel")
     set(MINSIZE ON)
@@ -27,10 +34,6 @@ option(ENABLE_CCACHE "Use ccache as compiler launcher, when available." ON)
 
 # cosmetics
 option(FORCE_COLORS "Force colored compiler output." OFF)
-
-# Build type options
-set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo" "")
-message(STATUS "Build type: ${CMAKE_BUILD_TYPE}")
 
 set(CMAKE_VISIBILITY_INLINES_HIDDEN ON)
 set(CMAKE_CXX_VISIBILITY_PRESET hidden)  # override to "default" for shared libs

--- a/cmake/XciBuildOptions.cmake
+++ b/cmake/XciBuildOptions.cmake
@@ -15,13 +15,6 @@ option(FAIL_ON_WARNINGS "Treat warnings as errors. Applies also to clang-tidy." 
 option(ENABLE_TIDY "Run clang-tidy on each compiled file, when available." OFF)
 option(ENABLE_IWYU "Run iwyu (Include What You Use) on each compiled file, when available." OFF)
 
-# default build type + possible values for cmake gui
-if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
-    set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build." FORCE)
-endif()
-set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
-message(STATUS "Build type: ${CMAKE_BUILD_TYPE}")
-
 # optimizations
 if(CMAKE_BUILD_TYPE STREQUAL "MinSizeRel")
     set(MINSIZE ON)

--- a/cmake/XciBuildType.cmake
+++ b/cmake/XciBuildType.cmake
@@ -1,0 +1,19 @@
+# I'd like to have Release as default build type on all platforms, so it's consistent with Conan.
+#
+# Without this workaround, it was quite wild:
+# * Mac, Linux: default is empty string, can be overridden using a snippet from [1]
+# * Windows with MSVC: default is Debug, override is more complicated, see [2]
+#
+# References:
+# [1] https://www.kitware.com/cmake-and-the-default-build-type/
+# [2] https://gitlab.kitware.com/cmake/cmake/-/issues/19401
+
+# set default build type to Release
+get_property(IS_MULTICONFIG GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+if (NOT IS_MULTICONFIG AND NOT DEFINED CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build.")
+endif()
+
+# set possible build types for cmake gui
+set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+message(STATUS "Build type: ${CMAKE_BUILD_TYPE}")

--- a/conanfile.py
+++ b/conanfile.py
@@ -57,11 +57,11 @@ class XcikitConan(ConanFile):
         "text": True,
         "widgets": True,
 
-        # Also build and install:
-        "tools": False,
-        "examples": False,
-        "tests": False,
-        "benchmarks": False,
+        # Also build and package:
+        "tools": True,
+        "examples": True,
+        "tests": True,
+        "benchmarks": True,
 
         # System dependencies (instead of Conan):
         "system_fmt": False,

--- a/src/xci/core/string.cpp
+++ b/src/xci/core/string.cpp
@@ -191,8 +191,10 @@ std::string to_lower(std::string_view str)
 
 std::u32string to_utf32(string_view utf8)
 {
+#ifdef __clang__
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#endif
     std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t> convert_utf32;
     try {
         return convert_utf32.from_bytes(utf8.data(), utf8.data() + utf8.size());
@@ -200,15 +202,19 @@ std::u32string to_utf32(string_view utf8)
         log::error("to_utf32: Invalid UTF8 string: {} ({})", utf8, e.what());
         return {};
     }
+#ifdef __clang__
 #pragma clang diagnostic pop
+#endif
 }
 
 
 template <class Elem>
 std::string _to_utf8(std::basic_string_view<Elem> wstr)
 {
+#ifdef __clang__
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#endif
     std::wstring_convert<std::codecvt_utf8<Elem>, Elem> convert;
     try {
         return convert.to_bytes(wstr.data(), wstr.data() + wstr.size());
@@ -216,7 +222,9 @@ std::string _to_utf8(std::basic_string_view<Elem> wstr)
         log::error("to_utf8: Invalid UTF16/32 string ({})", e.what());
         return {};
     }
+#ifdef __clang__
 #pragma clang diagnostic pop
+#endif
 }
 
 std::string to_utf8(std::u16string_view wstr) { return _to_utf8(wstr); }
@@ -228,11 +236,15 @@ std::string to_utf8(std::wstring_view wstr) { return _to_utf8(wstr); }
 
 std::string to_utf8(char32_t codepoint)
 {
+#ifdef __clang__
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#endif
     std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t> convert_utf32;
     return convert_utf32.to_bytes(codepoint);
+#ifdef __clang__
 #pragma clang diagnostic pop
+#endif
 }
 
 

